pkg/apiserver/registry/networkpolicy/networkpolicy/rest.go

```diff
@@ -61,28 +61,33 @@ func (r *REST) NewList() runtime.Object {
 
 func (r *REST) Get(ctx context.Context, name string, options *v1.GetOptions) (runtime.Object, error) {
 	ns, ok := request.NamespaceFrom(ctx)
-	if !ok || len(ns) == 0 {
-		return nil, errors.NewBadRequest("Namespace parameter required.")
-	}
-	key := k8s.NamespacedName(ns, name)
-	networkPolicy, exists, err := r.networkPolicyStore.Get(key)
-	if err != nil {
-		return nil, errors.NewInternalError(err)
-	}
-	if !exists {
-		return nil, errors.NewNotFound(networking.Resource("networkpolicy"), name)
+	networkPolicies := r.networkPolicyStore.List()
+	list := new(networking.NetworkPolicyList)
+	list.Items = make([]networking.NetworkPolicy, 0)
+	cnt := 0
+	for i := range networkPolicies {
+		policy := networkPolicies[i].(*types.NetworkPolicy)
+		if (len(name) == 0 || name == policy.Name) && (!ok || len(ns) == 0 || (ok && policy.Namespace == ns)) {
```

Comment:
I think `!ok` can never happen? The framework requires namespace parameter for namespace scoped query?
If only namespace is provided, it should call `List`

Reply:
Deleted.